.github/workflows/rust.yml

```diff
@@ -26,3 +26,43 @@ jobs:
       run: cargo test --verbose
     - name: Doc
       run: cargo doc
+
+  cross-linux:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target:
+        - arch: 'aarch64'
+          rust: 'aarch64-unknown-linux-gnu'
+          os: 'aarch64-linux-gnu'
+        - arch: 'arm'
+          rust: 'armv7-unknown-linux-gnueabihf'
+          os: 'arm-linux-gnueabihf'
+        - arch: 'i386'
+          rust: 'i686-unknown-linux-gnu'
+          os: 'i686-linux-gnu'
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install Rust toolchain
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        override: true
+        components: rustfmt, clippy
+        target: ${{ matrix.target.rust }}
+    - name: Install QEMU and toolchain
+      run: |
+        sudo apt update
+        sudo apt -y install qemu-user qemu-user-static gcc-${{ matrix.target.os }} binutils-${{ matrix.target.os }} binutils-${{ matrix.target.os }}-dbg
+    - name: Append .cargo config
+      run: |
+        echo "[target.${{ matrix.target.rust }}]" >> ~/.cargo/config
+        echo "runner = \"qemu-${{ matrix.target.arch }} -L /usr/${{ matrix.target.os }}/\"" >> ~/.cargo/config
+        echo "linker = \"${{ matrix.target.os }}-gcc\"" >> ~/.cargo/config
+    - name: Build
+      run: cargo build --target ${{ matrix.target.rust }}
+    - name: Run tests
+      run: cargo test --target ${{ matrix.target.rust }}
+    - name: Doc
+      run: cargo doc --target ${{ matrix.target.rust }}
```

Comment:
Another option, if you want, is to use actions-rs/cargo with the [`use-cross` option](https://github.com/actions-rs/cargo#cross-compilation) enabled. That makes the whole thing easier, and you don't have to install the dependencies or anything, and you only have to use the rust target triple in the matrix.

Reply:
Honestly, I don't know. I have looked at cross in the past and while it does look more convenient, I don't like its dependency on Docker/Podman at all.

This instead reflects my actual workflow which is to use rustup to install a Rust toolchain for the specific target, run the cargo commands with the specified target, and then run it in using QEMU userspace emulation for that target, which I agree is more involved, but at the same time I know what is going on exactly.

Also, I simply have more experience with QEMU userspace emulation, because I have used that for things outside of Rust like preparing a Linux image for AArch64 using chroot.

Maybe it offers better performance for Github CI? I am not sure, as at the end of the day I am not too familiar with cross, but that could be a good reason to use it for Github CI.